src/test/java/com/nimiq/NimiqClientTest.java

```diff
@@ -0,0 +1,467 @@
+
+        final TransactionReceipt receipt = client.getTransactionReceipt(TX1_HASH);
+        assertNotNull(receipt);
+        assertEquals(BLOCK_NUMBER, receipt.getBlockNumber());
+        assertEquals(BLOCK_HASH, receipt.getBlockHash());
+        assertEquals(TX1_HASH, receipt.getTransactionHash());
+        assertEquals(0, receipt.getTransactionIndex());
+    }
+
+    @Test
+    public void testGetTransactionsByAddress() {
+        final int limit = 100;
+        final List<Transaction> transactions = client.getTransactionsByAddress(FAUCET_ADDRESS, limit);
+        assertEquals(limit, transactions.size());
+        transactions.forEach(tx -> {
+            assertTrue(FAUCET_ADDRESS.equals(tx.getFromAddress()) || FAUCET_ADDRESS.equals(tx.getToAddress()));
+        });
+    }
+
+    @Test
+    public void testMempoolContent() {
+        final String sender = getSenderAddress(100_00000);
+
+        // Send a test transaction first
+        final OutgoingTransaction outTx = createTransaction(sender, FAUCET_ADDRESS, 5_00000, MIN_FEE);
+        final String txHash = client.sendTransaction(outTx);
+
+        final List<Transaction> txsWithoutDetails = client.mempoolContent(false);
+        assertTrue(txsWithoutDetails.size() > 0);
+        assertTrue(txsWithoutDetails.stream().anyMatch(tx -> tx.getHash().equals(txHash)));
+        txsWithoutDetails.forEach(tx -> {
+            assertNotNull(tx.getHash());
+            assertNull(tx.getFrom());
+            assertNull(tx.getTo());
+            assertEquals(0, tx.getValue());
+            assertNull(tx.getBlockHash());
+            assertEquals(0, tx.getBlockNumber());
+        });
+
+        final List<Transaction> txsWithDetails = client.mempoolContent(true);
+        assertTrue(txsWithDetails.size() > 0);
+        assertTrue(txsWithDetails.stream().anyMatch(tx -> tx.getHash().equals(txHash)));
+        txsWithDetails.forEach(tx -> {
+            assertNotNull(tx.getHash());
+            assertNotNull(tx.getFrom());
+            assertNotNull(tx.getTo());
+            assertTrue(tx.getValue() > 0);
+            assertNull(tx.getBlockHash());
+            assertEquals(0, tx.getBlockNumber());
+        });
+    }
+
+    @Test
+    public void testMempool() {
+        final String sender = getSenderAddress(100_00000);
+
+        // Send several transactions with different fees
+        int[] fees = { 100, 200, 500 }; // per byte
+        IntStream.of(fees).forEach(fee -> {
+            client.sendTransaction(createTransaction(sender, FAUCET_ADDRESS, 3_00000, fee * MIN_FEE));
+        });
+
+        final Mempool mempool = client.mempool();
+        assertTrue(mempool.getTotal() >= fees.length);
+        final int total = Arrays.stream(mempool.getBuckets()).map(mempool::getNumberOfTransactions).sum();
+        assertEquals(total, mempool.getTotal());
+        IntStream.of(fees).forEach(fee -> {
+            assertTrue(mempool.getNumberOfTransactions(fee) > 0);
```

Comment:
Would it make sense to keep track of the TX IDs when sending the TX and check here that those TX are in the mempool?

Reply:
yes, I do it in the method above, in testGetMempoolContent() and testGetMempool() is about "fees per byte buckets" (I don't even understand why anyone would ever need such a method 🤔 )